cohortextractor2/query_engines/spark_dialect.py

```diff
@@ -106,6 +115,39 @@ class SparkDialect(HiveHTTPDialect):
         sqlalchemy.types.DateTime: SparkDateTime,
     }
 
+    # This function is only excercised when manuall running the tests against databricks
+    def create_connect_args(self, url):  # pragma: no cover
+        """Switch between generic phyive dbapi and databricks dbapi as needed.
+
+        We use the generic pyhive dbapi in local tests, but we use the databricks api
+        in integration tests and  production.
+
+        The only significate difference in API is the format of the connection
+        arguments, and that the databricks dbapi uses async thrift.
+        """
+
+        if "http_path" not in url.query:
+            # regular pyhive connection
+            return super().create_connect_args(url)
+
+        # databricks connection
+        #
+        # Switch to use databricks pyhive based-dbapi reather than the default
+        # This is not very clean, but its simple and it seems to work.
+        self.dbapi = databricks.sql
+
+        server_hostname = url.host
+        http_path = url.query["http_path"]
+
+        # TODO: handle access_token
```

Comment:
Should this TODO still be here?

Reply:
Yes. In community edition, we cannot use tokens. In production, they will almost certainly use tokens, and the kwargs will be different.

I can add that in this PR if you'd prefer, but I was going to do it in a later one, as this one's already larger than I wanted